src/layouts/CoreLayout/CoreLayout.js

```diff
@@ -1,19 +1,151 @@
-import React from 'react'
-import Header from '../../components/Header'
-import './CoreLayout.scss'
-import '../../styles/core.scss'
-
-export const CoreLayout = ({ children }) => (
-  <div id="core-layout-component" className='text-center'>
-    <Header />
-    <div className='core-layout__viewport'>
-      {children}
-    </div>
-  </div>
-)
+import React, { Component } from 'react';
+import * as lib from '../../lib';
+import Header from '../../components/Header';
+import './CoreLayout.scss';
+import '../../styles/core.scss';
+
+export default class CoreLayout extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      pos: this.props.initialPos,
+      dragging: false,
+      rel: null // position relative to the cursor
+    };
+  }
+
+  /*  we could get away with not having this (and just having the listeners on
+   our div), but then the experience would be possibly be janky. If there's
+   anything w/ a higher z-index that gets in the way, then you're toast,
+   etc.*/
+  componentDidUpdate(props, state) {
+    if (this.state.dragging && !state.dragging) {
+      document.addEventListener('mousemove', this.onMouseMove);
+      document.addEventListener('mouseup', this.onMouseUp);
+    } else if (!this.state.dragging && state.dragging) {
+      document.removeEventListener('mousemove', this.onMouseMove);
+      document.removeEventListener('mouseup', this.onMouseUp);
+    }
+  }
+
+  // calculate relative position to the mouse and set dragging=true
+  onMouseDown(e) {
```

Comment:
should all of this stuff really be on CoreLayout? Surely it would be better on the Table component.